### PR TITLE
Resolve issue 66 without .so files in history

### PR DESCRIPTION
**Pull request by dannytoomey**
Hi Witold,

Here's a new PR that resolves issue #66 with no .so files in the commit history. 

In the stub, I saw your had an estimate variable that appeared to be a list of length 3. Could you clarify what you had in mind for this? I didn't see an analogous estimate variable in the [example function](https://www.metafor-project.org/doku.php/plots:meta_analytic_scatterplot). Thanks!

Best regards,

Danny Toomey

### CODE REVIEW

**Comment by dannytoomey** [comment]
Don't merge yet, this does not pass build checks.

File: R/forest_plot.R [hunk 1]
```diff
@@ -136,3 +136,45 @@ forest_plot <- function(bg,
 
   do.call(forestplot, l)
 }
+
+#' Draw a bubble plot using ggplot syntax
```
[the remark this answers is not in the document]
Comment (dannytoomey) [reply R/forest_plot.R:140]: Got it, added this

File: R/forest_plot.R [hunk 2]
```diff
@@ -136,3 +136,45 @@ forest_plot <- function(bg,
...
+#' @examples
+#' 
+#' covariates <- rnorm(8)
+#' bg <- baggr(schools)
+#' bubble_plot(bg,covariates)
+#'
+#' @export
+bubble_plot <- function(bg, covariates, regression=TRUE, interval=0.95) {
+  data <- bg$data
+  data$se <- 1/(data$se^2)
```
[the remark this answers is not in the document]
Comment (dannytoomey) [reply R/forest_plot.R:173]: Yes, this only works for rubin models. Wrote tests for this for a couple model types.

File: R/forest_plot.R [hunk 3]
```diff
@@ -136,3 +136,50 @@ forest_plot <- function(bg,
...
+  data$mean <- as.data.frame(group_effects(bg, summary = TRUE, interval = interval)[,,1])$mean
+
+  group <- NULL
+  
+  ggplot2::ggplot(data, aes(x = covariates,y=mean)) + 
+  ggplot2::geom_point(aes(size=data$se)) +
+  ggplot2::scale_size_continuous(name = "1/SE^2") + 
+  ggplot2::xlab("Covariates") + ggplot2::ylab(bg$effects) +
+  {if(regression) ggplot2::geom_smooth(method="lm",level=interval, col="black")} +
+  {if(add_label) ggrepel::geom_text_repel(aes(x = covariates,y=mean,label=data$group),box.padding=1)}
```
Comment (dannytoomey) [inline R/forest_plot.R:183]: I added study labels so it reads a little easier.

File: R/forest_plot.R [hunk 4]
```diff
@@ -136,3 +136,45 @@ forest_plot <- function(bg,
...
+
+#' Draw a bubble plot using ggplot syntax 
+#'
+#' @param bg          A [baggr] object.
+#' @param covariates  Covariates for the [baggr] data. 
+#' @param regression  A logical value that sets an overlying linear regression and 
+#'                    confidence interval on or off.
+#' @param interval    The confidence interval used in the linear regression, 
+#'                    given as a value between 0 and 1.
+#' @return A bubble plot of 1/SE^2 for Tau as a function of the given covariates with
```
[the remark this answers is not in the document]
Comment (dannytoomey) [reply R/forest_plot.R]: I misunderstood what this plot shows. I updated the docs with more details and better explained what this plot returns.

File: R/stanmodels.R
[the remark this answers is not in the document]
Comment (dannytoomey) [reply R/stanmodels.R:1]: Removed! And will remove from the other PR's

File: R/forest_plot.R [hunk 5]
```diff
@@ -136,3 +136,45 @@ forest_plot <- function(bg,
...
+  data <- bg$data
+  data$se <- 1/(data$se^2)
+
+  #if(!missing(estimate)){
+  #  fit_ribbon_data <- data.frame(x = seq(min(data[[covariates]]), 
+  #    max(data[[covariates]]), length = 100)) %>%
+  #    mutate(y = x*estimate[2], ymin = x*estimate[1], ymax = x*estimate[3])       
+  #}
+  if(regression){
+    ggplot(data, aes_string(x = covariates, y = "tau")) +
```
[the remark this answers is not in the document]
Comment (dannytoomey) [reply R/forest_plot.R]: Yes, this was my misunderstanding of what this plot represents. If I correctly understand this plot now, y is the mean treatment effect given by group effects and x is the covariate. The regression line predicts the mean treatment effect for the covariates. I updated the docs to explain this more thoroughly. Let me know if I'm still misunderstanding this plot!

**Comment by wwiecek** [comment]
Reminder to self: this was since then resolved elsewhere and I need to bring in that bubble plot function into the package (instead of using this solution). I think in the h2o paper.

**Comment by wwiecek** [comment]
I programmed a different solution now

**Pull request closed without merging**